src/cruds/device.py

```diff
@@ -218,14 +218,16 @@ async def get_historical_device_data_alarm(db: AsyncSession, device_id: str) ->
     Returns:
         [device.schema.DeviceHistoricalAlarm]: List of device historical alarm data
     """
+    #
     stmt = """
         SELECT
             IS_ALARM,
-            TO_CHAR(CREATED_AT, 'YYYY/MM/DD')
+            TO_CHAR(CREATED_AT, 'YYYY/MM/DD') AS DATE,
+            TO_CHAR(CREATED_AT, 'HH24:MI') AS HOUR
         FROM ALARM
         WHERE
             DEVICE_ID = :device_id
-        ORDER BY CREATED_AT
+        ORDER BY DATE
```

Comment:
I think you need to also order by `HOUR` too.

Reply:
Good catch!

Updated to: 
ORDER BY DATE, HOUR;

Tested and pushed.